biz.aQute.repository/src/aQute/maven/provider/Releaser.java

```diff
@@ -71,24 +71,30 @@ protected void updateMetadata() throws Exception, InterruptedException {
 		int n = 0;
```

Comment:
Seems like there is still an NPE risk here. You now alway call `updateMetadata` and it always calls `isUpdateProgramMetadata` whose implementation on `Releaser` assumes `repo` is non-null. I think you also need to fix `Releaser.isUpdateProgramMetadata` to guard against repo being null.

Reply:
no because the implementation overrides that method:
```
	protected boolean isUpdateProgramMetadata() {
		return true;
	}
```

Reply:
now i did consider that maybe the Snapshot releaser should just provide a completely different path for the non-repo case as opposed to having all these null checks when driving through the same logical path.

Reply:
ok... now I see what you mean...

Reply:
maybe there should be a new path for SnapshotReleaser with no repo?

Reply:
That is possible. I thought of something like that. But I am not sure that is just too risky.

Reply:
Also, I think you can release a non-snapshot version to a local only repository and so Releaser will be used with repo==null. So the fix cannot be only in SnapshotReleaser.

Reply:
I think you only need up update the `Releaser.isUpdateProgramMetadata` method to handle repo==null. I dont think it is a big change.